CMakePresets.json

```diff
@@ -0,0 +1,202 @@
+{
```

Comment:
@DarkWanderer are these presets just for running the unit tests? Or will they affect the main project  too?  i.e., are these the same changes as in the other PR that @mschweiger had some questions about?

Reply:
Yes, these are same presets. `Visual Studio` generator has an issue of binaries being placed into e.g. `Modules\Release` directory instead of `Modules`. This breaks running of tests from build directory completely. Enabling presets was the quickest way to switch to Ninja generator

Is it a blocker for merging this PR?

Reply:
@DarkWanderer TBH I don't know how big of a blocker the presets are -- I just presumed there are *a* blocker for the other PR since the PR had sat open for 22 days. It's really up to @mschweiger since he's the product owner, and I really don't want to step on anyone's toes, yours or his. 😄 

Great work on all these tests! It's much appreciated!